photosynthesis_metrics/tv.py

```diff
@@ -22,7 +22,9 @@ def total_variation(x: torch.Tensor, size_average: bool = True, reduction_type:
         https://www.wikiwand.com/en/Total_variation_denoising
         https://remi.flamary.com/demos/proxtv.html
     """
-    assert x.dim() == 4, f'Expected 4D tensor, got {x.size()}'
+    _validate_input(x)
+    x = _adjust_dimensions(x)
```

Comment:
Tensor of `(3, 3, 256, 256, 2)` size will pass that check. SSIM can deal with 2/3/4/5D, while other losses deal only with 2/3/4D tensors. Let's discuss how we can resolve it in a better way than one more `assert`.

Reply:
We can add a parameter to the `_validate_input` function, which indicates support for 5D. 
The interface for TV will be like:
`_validate_input(x, supports_5d=False)`

Reply:
@denproc what do you think?

Reply:
Looks like a plan 👌🏻. +1